### PR TITLE
feat(rbac): add conditional access button and sidebar

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Story: https://issues.redhat.com/browse/RHIDP-1639

This PR adds plugins conditional access button which when clicked opens the sidebar with the list of applicable rules.

https://github.com/janus-idp/backstage-plugins/assets/20724543/20c156f1-88b7-450d-a5a6-a8e1ddf167ec

### CODE REVIEW

File: plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx [hunk 1]
```diff
@@ -47,73 +57,109 @@ export const PermissionPoliciesFormRow = ({
...
+              variant="outlined"
+              placeholder="Select a permission"
+              error={!!permissionError}
+              helperText={permissionError ?? ''}
+              onBlur={handleBlur}
+              required
+            />
+          )}
+        />
+        <IconButton
```
Comment (invincibleJai) [inline plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx:125]: good to comment the CTA now with proper comment as feature is not enabled

Comment (invincibleJai) [reply plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx:126]: HI @divyanshiGupta , shall we keep the CTA as well commented for now?

Comment (divyanshiGupta) [reply plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx:126]: @invincibleJai not sure if that is really required. Also if I comment this out I will have to comment out the related tests as well. Lmk if I should do that.

**Comment by invincibleJai** [comment]
@divyanshiGupta overall looks good , i have observed on create flow if go on step to add permission policies and go back and revisit step add permission policies then the CTA for conditional policies is disabled 



https://github.com/janus-idp/backstage-plugins/assets/5129024/000d18e5-8906-4458-89d5-c0f1d66b73f6


and i see a warning , this shouldn't be related to this PR but take a look once

```
Material-UI: The value provided to Autocomplete is invalid.
None of the options match with `{"label":"","etag":"","type":"","ref":""}`.
You can use the `getOptionSelected` prop to customize the equality test.
```

File: plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx (hunk 1 above)
Comment (divyanshiGupta) [reply plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx]: Commented out the cta and kept the layout same as before.

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta overall looks good , i have observed on create flow if go on step to add permission policies and go back and revisit step add permission policies then the CTA for conditional policies is disabled
> 
>  RBACconditionalPolicy480.mov 
> and i see a warning , this shouldn't be related to this PR but take a look once
> 
> ```
> Material-UI: The value provided to Autocomplete is invalid.
> None of the options match with `{"label":"","etag":"","type":"","ref":""}`.
> You can use the `getOptionSelected` prop to customize the equality test.
> ```

Fixed the first issue.

Looked into the warning and opened a ticket - https://issues.redhat.com/browse/RHIDP-1775

**Comment by invincibleJai** [comment]
@divyanshiGupta can you rebase and update the PR again, thanks!!

File: plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.test.tsx [hunk 2]
```diff
@@ -67,4 +68,29 @@ describe('PermissionPoliciesFormRow', () => {
 
     expect(mockProps.onChangePlugin).toHaveBeenCalledWith('Plugin1');
   });
+
+  // eslint-disable-next-line jest/no-commented-out-tests
+  // it('opens sidebar on clicking conditional access button', async () => {
```
Comment (ciiay) [inline plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.test.tsx:73]: We can simply add an `x` to this test if we want to keep the code and skip it.
```
xit('opens sidebar on clicking conditional access button', async () => {...
```

File: plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx [hunk 3]
```diff
@@ -4,10 +4,15 @@ import { makeStyles } from '@material-ui/core';
 import IconButton from '@material-ui/core/IconButton';
 import TextField from '@material-ui/core/TextField';
 import Autocomplete from '@material-ui/lab/Autocomplete';
+// import ChecklistRtlIcon from '@mui/icons-material/ChecklistRtl';
+// import HelpOutlineIcon from '@mui/icons-material/HelpOutline';
 import RemoveIcon from '@mui/icons-material/Remove';
+// import Tooltip from '@mui/material/Tooltip';
```
Comment (ciiay) [inline plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx:10]: Do we need these comment outs?

**Review by ciiay — COMMENTED** [review]
Hi @divyanshiGupta , looks like clicking outside of the drawer doesn't close it. The rest works great for me.

**Comment by divyanshiGupta** [comment]
> Hi @divyanshiGupta , looks like clicking outside of the drawer doesn't close it. The rest works great for me.

@ciiay I have intentionally not enabled sidebar close on outside click as IMO the sidebar which is going to have a form should not get accidentally closed while editing it if the user clicks unintentionally outside. Will anyways confirm with UX. Thanks!

**Comment by divyanshiGupta** [comment]
> > Hi @divyanshiGupta , looks like clicking outside of the drawer doesn't close it. The rest works great for me.
> 
> @ciiay I have intentionally not enabled sidebar close on outside click as IMO the sidebar which is going to have a form should not get accidentally closed while editing it if the user clicks unintentionally outside. Will anyways confirm with UX. Thanks!

cc: @ShiranHi

**Comment by ShiranHi** [comment]
> @ciiay I have intentionally not enabled sidebar close on outside click as IMO the sidebar which is going to have a form should not get accidentally closed while editing it if the user clicks unintentionally outside. Will anyways confirm with UX. Thanks!

 That is correct, no need to close it when clicking outside of the side panel

**Pull request merged**